tests/sparse_ops_test.py

```diff
@@ -315,6 +319,107 @@ def test_matmul(self, shape, dtype, Obj, bshape):
 
     self.assertAllClose(M @ x, Msp @ x, rtol=MATMUL_TOL)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_{}".format(jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype}
+      for shape in [(5, 8), (8, 5), (5, 5), (8, 8)]
+      for dtype in jtu.dtypes.floating + jtu.dtypes.complex))
+  def test_coo_todense_ad(self, shape, dtype):
+    rng = rand_sparse(self.rng(), post=jnp.array)
+    M = rng(shape, dtype)
+    data, row, col = sparse_ops.coo_fromdense(M, nnz=(M != 0).sum())
+    f = lambda data: sparse_ops.coo_todense(data, row, col, shape=M.shape)
+
+    # Forward-mode
+    primals, tangents = api.jvp(f, [data], [jnp.ones_like(data)])
```

Comment:
Might be easier to use `jacfwd` and `jacrev` in place of `jvp` and `vjp`, then compare all resulting entries.

Perhaps useful for all of the sparse work (AD aside) is comparing against a non-sparse reference computation (on the non-zero entries).

We can also test higher-order AD (e.g. forward-on-reverse), for instance with `jax.hessian`.